lib/Int64.fram

```diff
@@ -0,0 +1,71 @@
+pub method equal {self : Int64} = (extern dbl_eqInt64  : Int64 -> Int64 -> Bool) self
+pub method neq   {self : Int64} = (extern dbl_neqInt64 : Int64 -> Int64 -> Bool) self
+pub method gt    {self : Int64} = (extern dbl_gtInt64  : Int64 -> Int64 -> Bool) self
+pub method lt    {self : Int64} = (extern dbl_ltInt64  : Int64 -> Int64 -> Bool) self
+pub method ge    {self : Int64} = (extern dbl_geInt64  : Int64 -> Int64 -> Bool) self
+pub method le    {self : Int64} = (extern dbl_leInt64  : Int64 -> Int64 -> Bool) self
+
+pub method toString = (extern dbl_int64ToString : Int64 -> String) self
+
+pub method neg {self : Int64} = (extern dbl_negInt64 : Int64 -> Int64) self
+pub method add {self : Int64} = (extern dbl_addInt64 : Int64 -> Int64 -> Int64) self
+pub method sub {self : Int64} = (extern dbl_subInt64 : Int64 -> Int64 -> Int64) self
+pub method mul {self : Int64} = (extern dbl_mulInt64 : Int64 -> Int64 -> Int64) self
+
+pub let zero = 0L
+pub let one  = 1L
+pub let minus_one = (1L).neg
+pub method succ {self : Int64} = self + one
+pub method pred {self : Int64} = self - one
+
+pub method div {`re : {type X} -> Unit ->[|_] X} (n : Int64) =
+  if n.equal zero then `re ()
+  else (extern dbl_divInt64 : Int64 -> Int64 -> Int64) self n
+
+pub method mod {`re : {type X} -> Unit ->[|_] X} (n : Int64) =
+  if n.equal zero then `re ()
+  else (extern dbl_modInt64 : Int64 -> Int64 -> Int64) self n
+
+pub method land = (extern dbl_andInt64 : Int64 -> Int64 -> Int64) self
+pub method lor  = (extern dbl_orInt64  : Int64 -> Int64 -> Int64) self
+pub method lxor = (extern dbl_xorInt64 : Int64 -> Int64 -> Int64) self
+
+pub method shiftl  = (extern dbl_lslInt64 : Int64 -> Int64 -> Int64) self
+pub method shiftr  = (extern dbl_lsrInt64 : Int64 -> Int64 -> Int64) self
+pub method ashiftr = (extern dbl_asrInt64 : Int64 -> Int64 -> Int64) self
+
+pub let int64_of_int = (extern dbl_int64ToInt : Int -> Int64)
```

Comment:
1. We agreed to use camelCase naming convention.
2. Maybe we could define a method toInt64 for type Int instead of this function.